### PR TITLE
Updated README

### PR DESCRIPTION
**Pull request by janettech**
Hello,

I have updated the README of Gaphas. I have reviewed and edited the file for :
1. Spelling and grammar
2. Edited the hyperlinks for virtual env

I do have comments on this line--
The constraint solver can also be used to keep constraints
contained within the item true, for example, to make sure a box maintains its
rectangular shape. (the word "true" is confusing).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -736,8 +739,9 @@ Move the item and lock the position of any element that's located at the same
 position.
 
 - item (`gaphas.item.Item`): The item to move.
-- view (`gaphas.view.View`): The view with guides to to use for move
-coordinates.
+- view (`gaphas.view.View`): The view with guides to use for move
+
```
Comment (danyeaw) [inline README.md]: Why was an extra return added here and on line 762?

File: README.md [hunk 2]
```diff
@@ -835,20 +840,21 @@ rect = Rectangle(1, 1, 5, 5)
 
 #### Decorator: @AsyncIO
 
-Schedule an ide handler at a given priority.
+Schedule an idle handler at a given priority.
 
-- single (bool): Schedules the decorated function to only be called a signle
-time.
+- single (bool): Schedules the decorated function to be called only a single
+
```
Comment (danyeaw) [inline README.md]: Extra return?

File: README.md [hunk 3]
```diff
@@ -879,33 +887,43 @@ def a(self, x=0):
 Thanks goes to these wonderful people ([emoji key](https://github.com/kentcdodds/all-contributors#emoji-key)):
 
 <!-- ALL-CONTRIBUTORS-LIST:START - Do not remove or modify this section -->
+
```
Comment (danyeaw) [inline README.md]: This section is autogenerated by the all-contributors tool, can we undo the modification?

File: README.md [hunk 4]
```diff
@@ -879,33 +887,43 @@ def a(self, x=0):
...
-    familiar with the codebase yet.
-2.  Fork [the repository](https://github.com/gaphor/gaphas) on
-    GitHub to start making your changes to the **master** branch (or
-    branch off of it).
-3.  Write a test which shows that the bug was fixed or that the feature
-    works as expected.
-4.  Send a pull request and bug the maintainers until it gets merged and
-    published. :smile:
+1. Check for open issues or open a fresh issue to start a discussion
+
```
Comment (danyeaw) [inline README.md]: Why were the extra lines added in this section?

**Comment by danyeaw** [comment]
> The constraint solver can also be used to keep constraints contained within the item true, for example, to make sure a box maintains it rectangular shape. (the word "true" is confusing).

What if we use the term [satisfied](https://en.wikipedia.org/wiki/Constraint_satisfaction) instead of true? Would that be more clear for you?

**Comment by janettech** [comment]
Hello Dan,

Thanks. I did have a feeling I did mess up with the indentation. Which markup editor did you use? I used MarkText and the content appearance was not the same as the README here. 

Let me know if you prefer a particular one and I will install that to edit your review comments.

**Comment by danyeaw** [comment]
@janettech If Marktext isn't working, I would recommend a code editor like VSCode. There is a [tutorial](https://sciwiki.fredhutch.org/compdemos/vscode_markdown_howto/) about using Markdown with VSCode. Jump on Gitter if you are having any issues.

**Comment by danyeaw** [comment]
@janettech How are things going?

File: README.md [hunk 5]
```diff
@@ -139,65 +143,63 @@ create_canvas(c, "Simple Gaphas App")
...
+coordinate system with x and y position, for example, a (0, 0) point.
 Item.matrix is the transformation relative to the parent item of the Item, as
 defined in the Canvas.
 
 Handles can connect to Ports. A Port is a location (line or point) where a
 handle is allowed to connect on another item. The process of connecting
 depends on the case at hand, but most often involves the creation of some
 sort of constraint between the Handle and the item it is connecting to (see
-doc/ports.txt).
+doc/ports.rst).
```
[the remark this answers is not in the document]
Comment (janettech) [reply README.md]: Yes, this can be done.

File: README.md [hunk 6]
```diff
@@ -223,105 +225,96 @@ item, etc.
...
 in order to provide more complex behaviour.
 
 To make it easy, a DefaultTool has been defined which is a ToolChain instance
 with the tools added as follows:
 
-- ToolChain - Delegates to a set of individual tools and keeps track of which
-tool has grabbed the focus. This normally happens when the user presses a mouse
-button. Once this happens, the tool requests a "grab" and all events, like
-motion or button release, are sent directly to the focused tool.
+- ToolChain - Delegates to a set of individual tools and keeps track of which tool has grabbed the focus. This normally happens when the user presses a mouse button. Once this happens, the tool requests a "grab" and all events, like motion or button release, are sent directly to the focused tool.
```
[the remark this answers is not in the document]
Comment (janettech) [reply README.md]: I have installed VScode and I see that they are correctly placed in the editor. I am not sure if I am missing anything here.

File: README.md (hunk 2 above)
Comment (janettech) [reply README.md]: Done

File: README.md (hunk 1 above)
Comment (janettech) [reply README.md]: Done

File: README.md (hunk 3 above)
Comment (janettech) [reply README.md]: I didn't touch these lines. Not sure whats the issue. Can you check again and let me know what needs to be done. I just edited the summary lines now as there were line breaks.

File: README.md (hunk 4 above)
Comment (janettech) [reply README.md]: Removed it.

**Comment by janettech** [comment]
Hi Dan,

I have updated the changes. I have a doubt:

Now I have added to edit your review comments. I have clicked on the pull request again. Are you able to see the modified file? Still learning ....:)

Thanks!

**Comment by danyeaw** [comment]
@janettech This looks great, thanks so much for you contribution, I really appreciate it! :heart: :tada:

**Comment by danyeaw** [comment]
@allcontributors please add @janettech for doc

**Comment by amolenaar** [comment]
Thanks @janettech! I love how you improved the readme. :)

**Comment by janettech** [comment]
Thanks @danyeaw and @amolenaar for your kind words. Glad I could do a small bit in your project :)

**Pull request merged**